devnet/cmd/scale-down-ecs-service/service_scaler.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/aws/aws-sdk-go/service/ecs/ecsiface"
+)
+
+type serviceScaler struct {
+	tickerDuration time.Duration
+	cluster        string
+	ecs            ecsiface.ECSAPI
+}
+
+func newServiceScaler(tickerDuration time.Duration, cluster string, ecs ecsiface.ECSAPI) *serviceScaler {
+	return &serviceScaler{
+		tickerDuration: tickerDuration,
+		cluster:        cluster,
+		ecs:            ecs,
+	}
+}
+
+func (ss serviceScaler) findServiceArns(ctx context.Context, serviceRegex string) (serviceArns []*string, err error) {
+	r, err := regexp.Compile(serviceRegex)
+	if err != nil {
+		return
+	}
+
+	var lsi = &ecs.ListServicesInput{
+		Cluster: &ss.cluster,
+	}
+	for {
+		var lso *ecs.ListServicesOutput
+		lso, err = ss.ecs.ListServicesWithContext(ctx, lsi)
```

Comment:
```suggestion
		lso, err := ss.ecs.ListServicesWithContext(ctx, lsi)
```

Reply:
I don't want to shadow `err` here. I'm using the named `err` return param

Reply:
Sure, up to you! Although if it's the same top scope in the function, the returned `err` will be set, so that would save one line of declaration.

By the way, maybe you have to do that due to using naked returns :wink: If you don't you can return some locally scoped error named `err` or something else.